src/fri/naive.md

```diff
@@ -0,0 +1,157 @@
+- the number of rounds $k=log(d)$ to derive the final polynomial implies the expected upper bound degree $d$
+- the same final value can be re-used to check consistency of the evaluation claims across layers
+
+We will refer this folding example again with some actual evaluations to demonstrate the interesting relationship among them.
+
+## What is the problem?
+
+Using the notation from [what is proving](https://nmohnblatt.github.io/zk-jargon-decoder/intro_to_zk/what_is_proving.html), we define the following:
+
+$$
+\mathcal{L}(\mathcal{R}_{\text{consistent-layers}})
+$$
+
+for instance, $\left(a_1, w_1\right) \in \mathcal{R}$, where $a_1$ can represent a FRI query and constant value from $p_2$, while $w_1$ represents the corresponding evaluations across layers.
+
+So the problem is to do the check to see if $a_1$ is in $\mathcal{L}$. In other words, the verifier checks if the prover can provide $w_1$ satisfying the query $a_1$.
+
+## naive FRI
+
+Let's assume prover is honest guy in this naive version of FRI. The goal is for verifier to do fast low degree test using the proof from prover. 
+
+### Hard honest work by prover
+
+Prover can use the folded polynomial to obtain the evaluations for each layers. These evaluations of layers can serve as a proof for the verifier to sample for their consistency across layers.
+
+Using the same polynomial example shown earlier, here are two simple calculation examples.
+
+For evaluation point at *1*:
+
+$$
+\begin{align} 
+\text{Layer 1:} & \\
+x_0&=1 \\
+p_0(x_0) &= 1 + 2x_0 + 3x_0^2 \\ 
+p_0(1) &= 6 \\ 
+p_0(-1) &= 2 \\\\
+\text{Layer 2:} & \\
+x_1&=x_0^2 \\
+p_1(x_1) &= 3 + 3x_1 \\
+p_1(1^2) &= 6 \\ 
+p_1(-1^2) &= 0 \\\\
+\text{Layer 3:} & \\
+x_2&=x_1^2 \\
+p_2(x_2) &= 6 \\
+p_2((1^2)^2) &= 6 \\\\
+\end{align}
+$$
+
+Similarly, for evaluation point at *2*:
+
+$$
+\begin{align} 
+\text{Layer 1:} & \\
+x_0&=2 \\
+p_0(x_0) &= 1 + 2x_0 + 3x_0^2 \\ 
+p_0(2) &= 17 \\ 
+p_0(-2) &= 9 \\\\
+\text{Layer 2:} & \\
+x_1&=x_0^2 \\
+p_1(x_1) &= 3 + 3x_1 \\
+p_1(2^2) &= 15 \\ 
+p_1(-2^2) &= 9 \\\\
+\text{Layer 3:} & \\
+x_2&=x_1^2 \\
+p_2(x_2) &= 6 \\
+p_2((2^2)^2) &= 6 \\\\
+\end{align}
+$$
+
+
+
+In this manner, the prover can keep calculating for the whole domain. Then they would response the verifier query with the evaluations across layers for checking if an instance $a_i$ corresponding to the query is in the $\mathcal{L}\left(\mathcal{R}_{\text{consistent-layers }}\right)$
+
+### Consistency check
+
+Supposed the verifier samples the evaluation point *2* as demonstrated in the example above, these evaluations across layers for that point will be provided by the prover.
+
+Based on the relationship implied from the equation of folding polynomial, the verifier can check the consistency among the evaluations between layers. Let's review and put this equation into use:
+
+$$p_i\left(x^2\right)=\frac{p_{i-1}\left(x\right)+p_{i-1}\left(-x\right)}{2}+\beta_i \frac{p_{i-1}\left(x\right)-p_{i-1}\left(-x\right)}{2 x}$$
```

Comment:
here again the equation falls out of the sky. IMO a first naive FRI that doesn't do that, and use g and h would be easier. And then you can move to an optimized version of FRI that gets rid of g and h. For example:

Notice that if the verifier chooses the points they want to query carefully, for example $v$ and $-v$, they don't need to directly query $g$ and $h$ but can infer their evaluations instead. This is because by querying only $p_0$ at these two points the verifier just has to solve this system of equation:

* $p_0(v) = g_0(v^2) + v \cdot h_0(v^2)$
*. $p_0(-v) = g_0(v^2) - v \cdot h_0(v^2)$

which is possible as we have [2 linearly independent equations of 2 unknowns](https://en.wikipedia.org/wiki/System_of_linear_equations)

then the next layer can be obtained locally without any further queries by the verifier: $p_1(v) = g_0(v) + \beta \cdot h_0(v)$

Reply:
IIUC, the $g_0(v^2)$ and $h_0(v^2)$ can be obtained by solving the system of equation.

Then, does it mean that, in the equation $p_1(v) = g_0(v) + \beta \cdot h_0(v)$, $g_0(v)$ and $h_0(v)$ actually represent the same values of $g_0(v^2)$ and $h_0(v^2)$ respectively? I am confused by the use of $v$ and $v^2$ between the $p_0$ and $p_1$.

Another question is, in this method, in order for the verifier to check $p_2(v)$, do they require prover to send $p_1(-v)$, so they can solve for the $g_1(v)$ and $h_1(v)$?

Reply:
Yeah, the ethstark paper seems saying the prover will send the $p_1(-v)$.

Reply:
yeah at each layer you query both the evaluation at v and -v, BUT as an optimization the prover commits to both these optimization under the same leaf in the merkle tree